### ISSUE TITLE
Feature Request: Allow dynamic OKS sigma values for pose evaluation metrics via dataset YAML

### ISSUE THREAD

**Issue by cosmo-gb** [ultralytics/ultralytics#24349]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Following up on @glenn-jocher's comment in #23843, I am proposing a solution to start cleaning up the hardcoded `OKS_SIGMA` logic, starting with the evaluation metrics.

Currently, `init_metrics` in `ultralytics/models/yolo/pose/val.py` defaults to COCO sigmas or uniform sigmas. I propose allowing users to pass these dynamically via the dataset YAML.

The updated logic in `val.py` would look like this:

```python
# Check if custom OKS sigmas are provided in the dataset YAML
if 'kpt_oks_sigmas' in self.data:
    self.sigma = np.array(self.data['kpt_oks_sigmas'])
else:
    self.sigma = OKS_SIGMA if is_pose else np.ones(nkpt) / nkpt
```

### Use case

```markdown
When training and evaluating pose models on custom datasets (e.g., non-human subjects, animals, or industrial objects), the default COCO-based human sigmas are not applicable. 

As seen in previous community questions (e.g., #16141, #22527, and #23843), users frequently struggle with applying custom sigma values, often resorting to manually monkeypatching the source code. Allowing these values to be passed dynamically via the dataset YAML provides much-needed flexibility without breaking existing workflows.
```

### Additional

**Note regarding Training Loss & Tests:** As Glenn mentioned in #23843, `v8PoseLoss` also needs this cleanup. To keep PRs small and manageable as requested, I intentionally kept this initial PR focused solely on `val.py` for evaluation metrics. 

I already have the code ready on my fork and I am ready to submit the PR as soon as you give the green light.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cosmo-gb, thank you for your interest in Ultralytics 🚀 and for the clear feature request around dynamic OKS sigma values for pose evaluation metrics via dataset YAML. We also appreciate that you’re willing to contribute a PR 🙌

This is an automated response to help get things moving quickly, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this becomes a 🐛 Bug Report during implementation or testing, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks as well for outlining the proposed scope clearly and for keeping the change focused on evaluation metrics first ✅ That kind of incremental PR is very helpful for review.

**Comment by glenn-jocher** [comment]
Yes, please open a small PR for `PoseValidator.init_metrics()` to support optional `kpt_oks_sigmas` from the dataset YAML, keeping it eval-only for now, validating the list length against `kpt_shape[0]`, and including a focused test plus a brief docs note without adding a new `default.yaml` argument.

**Comment by cosmo-gb** [comment]
Hi @glenn-jocher, just a quick heads-up that I've opened the requested PR here: #24381
I've included the eval-only support, the validation against kpt_shape, the unit test, and the docs update as discussed. All CI checks are green and the bot's feedback has been addressed. Thanks again for the clear guidance!

**Comment by glenn-jocher** [comment]
Thanks for opening [PR #24381](https://github.com/ultralytics/ultralytics/pull/24381) and keeping it tightly scoped to the eval path with the validation, test, and docs update; I’ll continue the review there.

**Comment by cosmo-gb** [comment]
Just commenting to keep this issue open and prevent the stale bot from closing it.

The implementation for the evaluation metrics is ready in PR #24381 and is currently awaiting final review. Once that is merged, I will open the follow-up PR to handle the dynamic sigma loss logic for training as discussed. Thanks!

**Comment by glenn-jocher** [comment]
Thanks for the follow-up — no need to bump this further; the remaining work is tracked in `#24381`, so please keep implementation updates there and open the training-side sigma change as a separate PR once the eval-only piece is merged.
